packages/cli/src/reporters/index.ts

```diff
@@ -0,0 +1,24 @@
+import { join } from 'node:path'
+import fse from 'fs-extra'
+import type { ResolvedUserConfig, UnlighthouseRouteReport } from '@unlighthouse/core'
+import { reportJsonSimple } from './jsonSimple'
+import { reportJsonExpanded } from './jsonExpanded'
+import type { ReportJsonExpanded, ReportJsonSimple } from './types'
+
+export function generateReportPayload(reporter: 'jsonExpanded', reports: UnlighthouseRouteReport[]): ReportJsonExpanded
```

Comment:
This was a nice way of overloading the function and still being sure the right one was called.